docs/proposals/client-settings.md

```diff
@@ -0,0 +1,395 @@
+    // +optional
+    Default *ClientSettingsPolicyConfig `json:"default,omitempty"`
+}
+
+type ClientSettingsPolicyStatus struct {
+    // Conditions describe the current conditions of the ClientSettingsPolicy
+    // +optional
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+type ClientSettingsPolicyConfig struct {
+    // MaxBodySize sets the maximum allowed size of the client request body.
+    // Setting size to 0 disables checking of client request body size.
+    // +optional
+    MaxBodySize *Size `json:"maxBodySize,omitempty"`
+
+    // BodyTimeout defines a timeout for reading client request body. The timeout is set only for a period between
+    // two successive read operations, not for the transmission of the whole request body.
+    // +optional
+    BodyTimeout *Duration `json:"bodyTimeout,omitempty"`
+
+    // KeepAlive defines the keep-alive settings.
+    // +optional
+    KeepAlive *KeepAlive `json:"keepAlive,omitempty"`
+}
+
+// KeepAlive defines the keepalive settings.
+type KeepAlive struct {
+    // Requests sets the maximum number of requests that can be served through one keep-alive connection.
+    // After the maximum number of requests are made, the connection is closed.
+    // +optional
+    Requests *int32 `json:"requests,omitempty"`
+
+    // Time defines the maximum time during which requests can be processed through one keep-alive connection.
+    // After this time is reached, the connection is closed following the subsequent request processing.
+    // +optional
+    Time *Duration `json:"time,omitempty"`
+
+    // Timeout defines the keep-alive timeouts.
+    // +optional
+    Timeout *KeepAliveTimeout `json:"timeout,omitempty"`
+
+    // Disable disables keep-alive connections for misbehaving browsers.
+    // +optional
+    Disable *DisableType `json:"disable"`
+}
+
+// KeepAliveTimeout defines the timeouts related to keep-alive client connections.
+type KeepAliveTimeout struct {
+    // ServerTimeout sets the timeout during which a keep-alive client connection will stay open on the server side.
+    // The zero value disables keep-alive client connections.
+    // +optional
+    ServerTimeout *Duration `json:"serverTimeout,omitempty"`
+
+    // HeaderTimeout sets the timeout in the "Keep-Alive: timeout=time" response header field.
+    // +optional
+    HeaderTimeout *Duration `json:"headerTimeout,omitempty"`
+}
+
+// DisableType is the type of browsers to disable keep-alive connections on.
+type DisableType string
+
+var (
+    // NoneDisableType enables keep-alive connections for all browsers.
+    NoneDisableType DisableType = "None"
+
+    // SafariDisableType disables keep-alive connections for Safari and Safari-like browsers on macOS and macOS-like
+    // operating systems.
+    SafariDisableType DisableType = "Safari"
+
+    // MSIEDisableType disables keep-alive connections for old versions of MSIE, once a POST request is received.
```

Comment:
since the docs https://nginx.org/en/docs/http/ngx_http_core_module.html#keepalive_disable mention msie6 (Internet Explorer 6), I wonder if this feature is relevant at all these days.

Reply:
Hrm. Do you think the disable feature is relevant? I wonder if we should just leave it out altogether. We can always add it later if a user requests it..